### DISCUSSION TITLE
associate not working on observer class

### DISCUSSION THREAD

**Discussion by metinbulak** [filamentphp/filament#5031] [Help]
I have user and menu models one to many associated.
I can not set user_id on updated or created function in the observer class, the user_id is going to be null on database.

User Model
```
     public function menus()
    {
        return $this->hasMany(Menu::class);
    }
```

Menu model
```
    public function user()
    {
        return $this->belongsTo(User::class);
    }
```

in the MenuObserver class

```
    public function updated(Menu $menu)
    {
        $menu->user()->associate(auth()->user());
        dd($menu);
    }
```
![image](https://user-images.githubusercontent.com/29355105/203151934-3f37e030-3f9b-46bd-a1dd-e583ae963b0e.png)

**Comment by zepfietje** [comment]
You might have to save the updated `$menu`?

**Reply by metinbulak** [reply]
when I try to save object model $menu, it saves to database but livewire component get timeout response on the admin ui.
Additionally, there is no save step in the docs [https://filamentphp.com/docs/2.x/admin/resources/getting-started#authorization](https://filamentphp.com/docs/2.x/admin/resources/getting-started#authorization)

```
public function creating(Post $post): void
{
    $post->user()->associate(auth()->user());
}
```

**Reply by zepfietje** [reply]
The `creating` method is different than `updated`. There's no need to call `save()` on the model in `creating`, since it will be saved after that lifecycle hook.

**Reply by metinbulak** [reply]
when I save $menu in the updated method, I get timeout on the admin ui

![image](https://user-images.githubusercontent.com/29355105/203328260-255d2b33-edfa-49b4-ba11-c2b7dd59d2a0.png)

**Reply by zepfietje** [reply]
Possibly since `updated` is called recursively then? Not sure.

**Reply by metinbulak** [reply]
I only set observer in the eventserviceprovider

```
class EventServiceProvider extends ServiceProvider
{
    public function boot()
    {
        Menu::observe(MenuObserver::class);
    }

}
```

**Reply by mitsyara** [reply]
did you fix this?
I have a very similar problem when I try to implement Model's Observer.
The data got saved to db, but I got the same error when the app just runs in a loop (I guess).
